ledger/participant-integration-api/src/main/scala/platform/store/dao/events/EventsTableFlatEventsRangeQueries.scala

```diff
@@ -156,14 +156,16 @@ private[events] object EventsTableFlatEventsRangeQueries {
       val witnessesWhereClause =
         sqlFunctions.arrayIntersectionWhereClause("flat_event_witnesses", party)
       QueryParts.ByArith(
-        read = (range, limitExpr) => SQL"""
-            select #$selectColumns, array[$party] as event_witnesses,
-                   case when submitters = array[$party] then command_id else '' end as command_id
+        read = (range, _) =>
```

Comment:
There's a reason `QueryParts.ByArith` provides `limitExpr`.
Instead of ignoring it, please convert it to `Option[Int]` and use it like `limitOpt.map(sqlFunctions.limitClause).getOrElse("")`.

Reply:
why didn't any tests fail if there is a dependency on this behaviour though?

Reply:
addressed